chainer/functions/loss/contrastive.py

```diff
@@ -72,8 +72,9 @@ def backward(self, indexes, grad_outputs):
         dist = chainer.functions.repeat(dist[:, None], x_dim, axis=1)
         # avoid division by zero, 1e-7 is sufficiently small value that can be
         # represented even in half precision
+        eps = 5e-3 if dist.dtype == 'float16' else 1e-7
```

Comment:
I might be missing something but the comment says that 1e-7 can be represented in fp16.
I think that according to your PR description, the  comment should be updated too :)

Thanks

Reply:
Updated comment.